xbmc/addons/AddonManager.cpp

```diff
@@ -497,7 +497,7 @@ void CAddonMgr::CheckAndInstallAddonUpdates(bool wait) const
   std::lock_guard<std::mutex> lock(m_installAddonsMutex);
   VECADDONS updates;
   GetAddonUpdateCandidates(updates);
-  InstallAddonUpdates(updates, wait, AllowCheckForUpdates::YES);
+  InstallAddonUpdates(updates, wait, AllowCheckForUpdates::NO);
```

Comment:
I am unsure about this change. This is called in the context of a migration and...repository addons can also be served from the repository that is being updated. There are cases in which you want your repository addon to be updated so that an update on that repository addon triggers further updates on your addons. Is this to fix some known issue or are you just guessing it might cause problems?

Reply:
this is really just a guess because of the `std::lock_guard` being set and that seems to be the root of the deadlock #18758 works around. we can set this pr on hold until an issue is reported (hopefully not)